File: threedollar-domain/src/main/java/com/depromeet/threedollar/domain/domain/storedelete/StoreDeleteRequestCreator.java
package com.depromeet.threedollar.domain.domain.storedelete;

import com.depromeet.threedollar.common.docs.ObjectMother;
import lombok.AccessLevel;
import lombok.NoArgsConstructor;

@ObjectMother
@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class StoreDeleteRequestCreator {

    public static StoreDeleteRequest create(Long storeId, Long userId, DeleteReasonType type) {
        return StoreDeleteRequest.of(storeId, userId, type);
    }

}

File: threedollar-domain-rds/src/testFixtures/java/com/depromeet/threedollar/domain/rds/user/domain/storedelete/StoreDeleteRequestCreator.kt
package com.depromeet.threedollar.domain.rds.user.domain.storedelete

import com.depromeet.threedollar.domain.rds.user.domain.TestFixture
import com.depromeet.threedollar.domain.rds.user.domain.store.DeleteReasonType
import com.depromeet.threedollar.domain.rds.user.domain.store.Store
import com.depromeet.threedollar.domain.rds.user.domain.store.StoreDeleteRequest

@TestFixture
object StoreDeleteRequestCreator {

    @JvmStatic
    fun create(
            store: Store,
            userId: Long,
            reasonType: DeleteReasonType
    ): StoreDeleteRequest {
        return StoreDeleteRequest.builder()
            .store(store)
            .userId(userId)
            .reason(reasonType)
            .build()
    }

}
